From 7aa63b43ffac09bc97cf05dd1b5370667d17fde2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Bartoszkiewicz?= <mbartoszkiewicz@gmail.com>
Date: Sat, 12 Jan 2013 23:05:52 +0100
Subject: [PATCH] core: do not make sockets dependent on lo

/sys/subsystem/net/devices/lo is never considered active, so sockets
with BindToDevice=lo would never be activated.
(cherry picked from commit 7d0c710d72f8a6e5c6909c65700aa088c53aebc6)
---
 src/core/socket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/socket.c b/src/core/socket.c
index fcbcdbe..d755040 100644
--- a/src/core/socket.c
+++ b/src/core/socket.c
@@ -308,7 +308,7 @@ static int socket_add_device_link(Socket *s) {
 
         assert(s);
 
-        if (!s->bind_to_device)
+        if (!s->bind_to_device || streq(s->bind_to_device, "lo"))
                 return 0;
 
         if (asprintf(&t, "/sys/subsystem/net/devices/%s", s->bind_to_device) < 0)
